### PR TITLE
fix(test): Add OIDC token refresh to int_test pipeline

### PR DESCRIPTION
**Pull request by c-ryan-k**
Running into token timeout issues all over the place, this script is a stop-gap until a new Az CLI version (2.60.0 currently) fixes issues refreshing OIDC federated tokens.

This adds a script that runs every 4 minutes to request a new federated token for the CLI, which runs in the background while the rest of the tests complete.

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

**Comment by c-ryan-k** [comment]
Running test here:
https://github.com/c-ryan-k/azure-iot-ops-cli-extension/actions/runs/8838708756/job/24270515723

File: .github/workflows/int_test.yml [hunk 1]
```diff
@@ -228,6 +228,16 @@ jobs:
           client-id: ${{ secrets.AZURE_CLIENT_ID }}
           tenant-id: ${{ secrets.AZURE_TENANT_ID }}
           subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      - name: "OIDC Token exchange service"
+        run: |
+          while true; do
+            token_request=$ACTIONS_ID_TOKEN_REQUEST_TOKEN
```
Comment (digimaun) [inline .github/workflows/int_test.yml:234]: are these env vars masked as secrets ?

Comment (digimaun) [reply .github/workflows/int_test.yml:234]: Though I see they are officially documented options https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers#requesting-the-jwt-using-environment-variables

Comment (c-ryan-k) [reply .github/workflows/int_test.yml:234]: These are injected by github into all runners that use OIDC: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers

We could add a secondary statement line to mask them if necessary, but I haven't seen that as a suggestion or recommendation yet

**Review by digimaun — APPROVED** [review]

**Pull request merged**